pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -32,79 +33,135 @@ import (
 }
 
-func validateKubeSchedulerProfile(path *field.Path, profile *config.KubeSchedulerProfile) field.ErrorList {
-	allErrs := field.ErrorList{}
+func validateKubeSchedulerProfile(path *field.Path, profile *config.KubeSchedulerProfile) []error {
+	var errs []error
 	if len(profile.SchedulerName) == 0 {
-		allErrs = append(allErrs, field.Required(path.Child("schedulerName"), ""))
+		errs = append(errs, field.Required(path.Child("schedulerName"), ""))
 	}
-	return allErrs
+	errs = append(errs, validatePluginConfig(path, profile)...)
+	return errs
 }
 
-func validateCommonQueueSort(path *field.Path, profiles []config.KubeSchedulerProfile) field.ErrorList {
-	allErrs := field.ErrorList{}
+func validatePluginConfig(path *field.Path, profile *config.KubeSchedulerProfile) []error {
+	var errs []error
+	m := map[string]interface{}{
+		"DefaultPreemption":           ValidateDefaultPreemptionArgs,
+		"InterPodAffinity":            ValidateInterPodAffinityArgs,
+		"NodeAffinity":                ValidateNodeAffinityArgs,
+		"NodeLabel":                   ValidateNodeLabelArgs,
+		"NodeResourcesFitArgs":        ValidateNodeResourcesFitArgs,
+		"NodeResourcesLeastAllocated": ValidateNodeResourcesLeastAllocatedArgs,
+		"NodeResourcesMostAllocated":  ValidateNodeResourcesMostAllocatedArgs,
+		"PodTopologySpread":           ValidatePodTopologySpreadArgs,
+		"RequestedToCapacityRatio":    ValidateRequestedToCapacityRatioArgs,
+		"VolumeBinding":               ValidateVolumeBindingArgs,
+	}
+
+	seenPluginConfig := make(sets.String)
+	for i := range profile.PluginConfig {
+		pluginConfigPath := path.Child("pluginConfig").Index(i)
+		name := profile.PluginConfig[i].Name
+		args := profile.PluginConfig[i].Args
+		if seenPluginConfig.Has(name) {
+			errs = append(errs, field.Duplicate(pluginConfigPath, name))
+		} else {
+			seenPluginConfig.Insert(name)
+		}
+		if validateFunc, ok := m[name]; ok {
+			// type mismatch, no need to validate the `args`.
+			if reflect.TypeOf(args) != reflect.ValueOf(validateFunc).Type().In(1) {
+				errs = append(errs, field.Invalid(pluginConfigPath.Child("args"), args, "has to match plugin args"))
+				return errs
+			}
+			in := []reflect.Value{reflect.ValueOf(pluginConfigPath.Child("args")), reflect.ValueOf(args)}
+			res := reflect.ValueOf(validateFunc).Call(in)
+			// It's possible that validation function return a Aggregate, just append here and it will be flattened at the end of CC validation.
+			if res[0].Interface() != nil {
+				errs = append(errs, res[0].Interface().(error))
+			}
+		}
+	}
+	return errs
+}
+
+func validateCommonQueueSort(path *field.Path, profiles []config.KubeSchedulerProfile) []error {
+	var errs []error
 	var canon config.PluginSet
+	var queueSortName string
+	var queueSortArgs runtime.Object
 	if profiles[0].Plugins != nil {
 		canon = profiles[0].Plugins.QueueSort
+		if len(profiles[0].Plugins.QueueSort.Enabled) != 0 {
+			queueSortName = profiles[0].Plugins.QueueSort.Enabled[0].Name
+		}
```

Comment:
let's validate that `len(QueueSort.Enabled) <= 1`

Reply:
what about this?
Please squash all commits when you add this.

Reply:
I am doubt that the validate can be added here, my understanding is that we can start a scheduler with a configuration file where "QueueSort" plugin is not necessary specified, it can be defer to the staging merging of plugin configuration from custom plugins

https://github.com/kubernetes/kubernetes/blob/46563b0abebbb00e21db967950a1343e83a0c6a2/pkg/scheduler/apis/config/types.go#L286

Reply:
```
=> 286:         p.QueueSort = mergePluginSets(p.QueueSort, customPlugins.QueueSort)
(dlv) p p.QueueSort.Enabled
[]k8s.io/kubernetes/pkg/scheduler/apis/config.Plugin len: 1, cap: 1, [
        {
                Name: "PrioritySort",
                Weight: 0,},
]
(dlv) p customPlugins.QueueSort.Enabled
[]k8s.io/kubernetes/pkg/scheduler/apis/config.Plugin len: 0, cap: 0, nil

```

Reply:
Add the validation here force us to specify a custom queue sort plugin in the config file always. Did I miss something?

Reply:
That why I said `<= 1` and not `== 1` :)

Reply:
Gotcha, give me one more min.

Reply:
Done